common/lcs/src/ser.rs

```diff
@@ -239,7 +251,8 @@ where
         Ok(())
     }
 
-    fn serialize_unit_struct(self, _name: &'static str) -> Result<()> {
+    fn serialize_unit_struct(mut self, name: &'static str) -> Result<()> {
+        self.enter_named_container(name)?;
         self.serialize_unit()
```

Comment:
I'm slightly worried about the maintenability of not having a `leave_named_container` in the recursive calls in `Serializer` (e.g. here). While I suspect that just manifests all your calls occur in terminal position (so that you only pop a stack frame when you're done), I'm wondering what would happen if that was to change (e.g. because you restructure seq / tuple serialization). Wouldn't you have to refactor every "named" method here, potentially leading to a bug if one of them is forgotten?
It's possible I may just be overthinking this.

Reply:
So, initially I tried adding a call `self.leave_named_container()` at the end and the borrow checker made it clear that it makes no sense to modify something after use :)

To make this PR, I searched for `name:` and changed the (5) callbacks. It definitely seems like the right approach to me.